src/main/java/org/reaktivity/reaktor/internal/agent/ElektronAgent.java

```diff
@@ -691,7 +706,8 @@ private ReadCounters newReadCounters(
     {
         final int localId = localId(routeId);
         final String nukleus = nukleus(localId);
-        return new ReadCounters(counters, nukleus, routeId);
+        final String tag = resolver.resolveTag(routeId);
+        return new ReadCounters(counters, nukleus, tag, routeId);
```

Comment:
Use `localAddress` instead of `nukleus`. This will give `http2#0.localhost:8080` instead of `http2.localhost:8080` when tag is `localhost:8080`, and `http2#0.7676da876af` when tag is `null`.

We have complete freedom on what to name `http2#0` during routing as it is just a string and not actually tied to the nukleus name.